abl/util/lockfile.py

```diff
@@ -60,7 +60,8 @@ def __enter__(self):
             else:
                 fcntl.flock(self.file, lock_flags)
         except IOError, e:
-            if e[0] == errno.EAGAIN:
+            error_code = errno.EACCES if is_windows else errno.EAGAIN
```

Comment:
So if you want to be precise you need a is_bsd (flock) for linux and bsds which tests against EAGAIN only, a is_windows which test EACCESS, and the rest who are posix but w/o a BSD flock which would test against EACCESS and EAGAIN